src/mesh/mesh.cpp

```diff
@@ -1066,24 +1081,9 @@ void Mesh::Initialize(int res_flag, ParameterInput *pin, ApplicationInput *app_i
 #pragma omp for
       for (int i = 0; i < nmb; ++i) {
         auto &pmb = block_list[i];
-        auto &pbval = pmb->pbval;
-        if (multilevel) pbval->ProlongateBoundaries(0.0, 0.0);
-        // TODO(JoshuaSBrown): Dead code left in for possible future extraction of
-        // primitives
-        //        int il = pmb->is, iu = pmb->ie, jl = pmb->js, ju = pmb->je, kl =
-        //        pmb->ks,
-        //           ku = pmb->ke;
-        //        if (pbval->nblevel[1][1][0] != -1) il -= NGHOST;
-        //        if (pbval->nblevel[1][1][2] != -1) iu += NGHOST;
-        //        if (pmb->block_size.nx2 > 1) {
-        //          if (pbval->nblevel[1][0][1] != -1) jl -= NGHOST;
-        //          if (pbval->nblevel[1][2][1] != -1) ju += NGHOST;
-        //        }
-        //        if (pmb->block_size.nx3 > 1) {
-        //          if (pbval->nblevel[0][1][1] != -1) kl -= NGHOST;
-        //          if (pbval->nblevel[2][1][1] != -1) ku += NGHOST;
-        //        }
-
+        if (multilevel) {
+          ProlongateBoundaries(pmb->meshblock_data.Get());
+        }
```

Comment:
I love that you put curly braces, safer I think!

Reply:
I have been burned so many times by leaving the braces out, trying to add code to the loop later, and getting unexpected behaviour because I forgot to add braces to the now multi-line loop.